pkg/controller/stackconfigpolicy/controller_test.go

```diff
@@ -400,11 +474,81 @@ func TestReconcileStackConfigPolicy_Reconcile(t *testing.T) {
 				assert.Equal(t, 1, policy.Status.Resources)
 				assert.Equal(t, 1, policy.Status.Ready)
 				assert.Equal(t, policyv1alpha1.ReadyPhase, policy.Status.Phase)
+				var esSecret corev1.Secret
+				// Verify the config secret created by the stack config policy controller
+				err = r.Client.Get(context.Background(), types.NamespacedName{
+					Namespace: "ns",
+					Name:      esv1.StackConfigElasticsearchConfigSecretName(esFixture.Name),
+				}, &esSecret)
+				assert.NoError(t, err)
+				elasticSearchConfigJSONData, err := json.Marshal(policy.Spec.Elasticsearch.Config)
+				assert.NoError(t, err)
+				secretMountsJSONData, err := json.Marshal(policy.Spec.Elasticsearch.SecretMounts)
+				assert.NoError(t, err)
+				assert.Equal(t, esSecret.Data[ElasticSearchConfigKey], elasticSearchConfigJSONData)
+				assert.Equal(t, esSecret.Data[SecretsMountKey], secretMountsJSONData)
```

Comment:
another helper method could help making this a little easier to follow because it would hide all the json marshalling business that is not relevant. Something like `assertExpectedESSecretContent` or so